Cython/Utility/ModuleSetupCode.c

```diff
@@ -2230,7 +2273,7 @@ static PyObject* __Pyx_PyCode_New(
     );
 
 done:
-    #if CYTHON_AVOID_BORROWED_REFS
+    #if CYTHON_AVOID_BORROWED_REFS || CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS
     Py_XDECREF(varnames_tuple_dedup);
```

Comment:
This seems incorrect, given the version specific condition above. If `CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS == 1` but `PY_[LIMITED_]VERSION_HEX < 3.13b1`, we'd kill a reference here.